File: 3074. Apple Redistribution into Boxes.py
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        capacity.sort()
        capacity.reverse()

        apple_sum = sum(apple)
        for i, cap in enumerate(capacity):
            apple_sum -= cap
            if apple_sum <= 0:
                return i + 1
        return -1

File: 3074. Apple Redistribution into Boxes.cpp
class Solution {
public:
    int minimumBoxes(vector<int>& apple, vector<int>& capacity) 
    {
        int apSum = 0, ans = 0;
        for(int v:apple) apSum += v;

        sort(capacity.rbegin(), capacity.rend());
        for(int &v : capacity)
        {
            ans += 1;
            apSum -= v;
            if(apSum <= 0) break;
        }
        return ans;
    }
};